Yoast/wordpress-seo
# Removal of license manager dependency

## Context


* The license manager isn't used anymore and the dependency can removed. This also mean that old code related to license manager behaviour can removed.

## Summary


This PR can be summarized in the following changelog entry:

* N/A - Removal of all license manager related code.

## Relevant technical choices:

* Decided to move logic from `WPSEO_Extensions` to `WPSEO_Addon_Manager` this makes it possible to deprecate the `WPSEO_Extensions`. Used the slugs for references.
* I think we can remove the class `WPSEO_License_Page_Manager`, but maybe we want some logic that adds the notices, but this should be based on the `WPSEO_Addon_Manager`

Todo:
* [x] Check if all addons still work
* [x] Convert `WPSEO_Extension` to an array and deprecate the `WPSEO_Extension` as well. There is one place where it is used now, no need to have a value object
* [x] Deprecate `WPSEO_License_Page_Manager` or convert it to an admin integration?

## Test instructions

This PR can be tested by following these steps:

* Make sure the licenses page still works. 
  * It should show the different addons. Have WooCommerce active to see the box for Yoast SEO for Woocommerce is visible
  * Have an addon enabled to the it's state being installed.

## UI changes

## Documentation

## Quality assurance


Related https://github.com/Yoast/wordpress-seo/issues/12476
